source/agora/consensus/EnrollmentManager.d

```diff
@@ -823,6 +823,34 @@ public class EnrollmentManager
         return PublicKey(this.key_pair.V[]);
     }
 
+    /***************************************************************************
+
+        Get the index of this node's enrollment in the sorted enrolled UTXOs
+
+        Returns:
+            the index of the enrollment, or ulong.max if this node is not
+            enrolled as a validator.
+
+    ***************************************************************************/
+
+    public ulong getIndexOfEnrollment () @safe nothrow
+    {
+        if (this.enroll_key == Hash.init)
+            return ulong.max;
+
+        Hash[] utxo_keys;
+        if (!this.validator_set.getEnrolledUTXOs(utxo_keys))
+            assert(0);
+
+        foreach (idx, key; utxo_keys)
+            if (key == this.enroll_key)
+                return idx;
+
+        log.fatal("Index of the node's enrollment not found. Enrollment: {}",
+            this.enroll_key);
+        assert(0);
+    }
+
```

Comment:
I think this index is already known by the validator without creating a new function:-
`public ulong getIndexOfValidator (const Height height, Point K) const nothrow @safe`

Reply:
The index of `getIndexOfValidator` is not the same as the index of `getIndexOfEnrollment` because the former is based on the order of public keys.
Thanks.

Reply:
Perhaps we should be using the same sorting criteria for validators for both block signing and slashing. I think it is easier if we keep it consistent. Perhaps this can be done in a seperate PR though.

Reply:
Raised an issue to update the validator index to be consistent.